### PR TITLE
optimizer hints: add the docs for hint SET_VAR

### PR DESCRIPTION
**Pull request by winoros**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

ref to https://github.com/pingcap/tidb/pull/45893

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: optimizer-hints.md [hunk 1]
```diff
@@ -75,6 +75,16 @@ SELECT /*+ QB_NAME(QB1) */ * FROM (SELECT * FROM t) t1, (SELECT * FROM t) t2;
 >
 > 上述例子中，如果指定的 `QB_NAME` 为 `sel_2`，并且不给原本 `sel_2` 对应的第二个查询块指定新的 `QB_NAME`，则第二个查询块的默认名字 `sel_2` 会失效。
 
+### SET_VAR(VAR_NAME=VAR_VALUE)
+
+`SET_VAR(VAR_NAME=VAR_VALUE)` 通过 Hint 的形式在在单条 SQL 的生命周期内改变系统变量的值。通过这个 Hint 可以控制大多数和优化器、执行器相关的变量行为。在[系统变量](/system-variables.md)章节中可以看到具体哪些变量支持该 Hint。
```
Comment (songrijie) [inline optimizer-hints.md:80]:
```suggestion
`SET_VAR(VAR_NAME=VAR_VALUE)` 能够将会话级系统变量，以 Hint 的形式在语句运行期间进行修改。语句执行完毕后，变量自动在所处的会话中恢复到原值。通过这个 Hint 可以修改一部分和优化器、执行器相关的变量行为，支持修改的系统变量请查看[系统变量](/system-variables.md)。强烈建议不要利用此 Hint 修改没有明确支持的变量，可能由此引发不可预知的行为。 
```

File: optimizer-hints.md [hunk 2]
```diff
@@ -75,6 +75,16 @@ SELECT /*+ QB_NAME(QB1) */ * FROM (SELECT * FROM t) t1, (SELECT * FROM t) t2;
...
+
+```sql
+SELECT /*+ SET_VAR(MAX_EXECUTION_TIME=1234) */ @@MAX_EXECUTION_TIME;
```
Comment (songrijie) [inline optimizer-hints.md:83]:
```suggestion
SELECT /*+ SET_VAR(MAX_EXECUTION_TIME=1234) */ @@MAX_EXECUTION_TIME;
SELECT @@MAX_EXECUTION_TIME;
```

File: optimizer-hints.md [hunk 3]
```diff
@@ -75,6 +75,16 @@ SELECT /*+ QB_NAME(QB1) */ * FROM (SELECT * FROM t) t1, (SELECT * FROM t) t2;
...
+```
+
+执行上述 SQL 会看到返回结果是 Hint 中设置的 1234，而不是变量 MAX_EXECUTION_TIME 的默认值。
```
Comment (songrijie) [inline optimizer-hints.md:86]:
```suggestion
执行上述 SQL 会看到返回结果是 Hint 中设置的 1234，而不是变量 MAX_EXECUTION_TIME 的默认值。再次执行时，又恢复了默认值。
```

**Comment by songrijie** [comment]
在 `system-variable.md` 的开头，加一句对这个 hint 的描述

- 对 `SESSION` 作用域变量的更改 ...
- 对 `GLOBAL` 作用域变量的更改 ...
```
- Hint [SET_VAR](/optimizer-hints.md#SET_VAR(VAR_NAME=VAR_VALUE)) 能够将会话级系统变量，以 Hint 的形式在语句运行期间进行修改。语句执行完毕后，变量自动在所处的会话中恢复到原值。"受 Hint [SET_VAR](/optimizer-hints.md#SET_VAR(VAR_NAME=VAR_VALUE)) 控制：是"，表示此变量支持用 Hint `SET_VAR()` 在运行时修改。  
```

**Review by songrijie — COMMENTED** [review]
LGTM

File: system-variables.md [hunk 4]
```diff
@@ -42,12 +42,17 @@ SET GLOBAL tidb_distsql_scan_concurrency = 10;
 >
 > 单位相同的多个变量可能会争夺同一组资源。
 
+## Optimizer Hint SET_VAR
+
+Hint [SET_VAR](/optimizer-hints.md#SET_VAR(VAR_NAME=VAR_VALUE)) 能够将会话级系统变量，以 Hint 的形式在语句运行期间进行修改。语句执行完毕后，变量自动在所处的会话中恢复到原值。"受 Hint [SET_VAR](/optimizer-hints.md#SET_VAR(VAR_NAME=VAR_VALUE)) 控制：是"，表示此变量支持用 Hint `SET_VAR()` 在运行时修改。
```
Comment (Oreoxmt) [inline system-variables.md:47]:
```suggestion
从 v7.4.0 开始，部分 `SESSION` 作用域的变量可以通过 [SET_VAR](/optimizer-hints.md#SET_VAR(VAR_NAME=VAR_VALUE)) Hint 在语句执行期间临时修改变量的值。当语句执行完成后，系统变量将在当前会话中自动恢复为原始值。通过这个 Hint 可以修改一部分与优化器、执行器相关的系统变量行为。文档中的变量有一个“是否受 Hint [SET_VAR](/optimizer-hints.md#SET_VAR(VAR_NAME=VAR_VALUE)) 控制”的说明，可以为“是”或者“否”。

- 对于受 Hint SET_VAR 控制的变量，你可以在语句中使用 `/*+ SET_VAR(...) */` 修改语句执行期间变量的值。
- 对于不受 Hint SET_VAR 控制的变量，你不能在语句中使用 `/*+ SET_VAR(...) */` 修改语句执行期间变量的值。

关于 SET_VAR Hint 的更多说明，参考 [SET_VAR](/optimizer-hints.md#SET_VAR(VAR_NAME=VAR_VALUE))。
```

File: system-variables.md [hunk 5]
```diff
@@ -42,12 +42,17 @@ SET GLOBAL tidb_distsql_scan_concurrency = 10;
...
+
+Hint [SET_VAR](/optimizer-hints.md#SET_VAR(VAR_NAME=VAR_VALUE)) 能够将会话级系统变量，以 Hint 的形式在语句运行期间进行修改。语句执行完毕后，变量自动在所处的会话中恢复到原值。"受 Hint [SET_VAR](/optimizer-hints.md#SET_VAR(VAR_NAME=VAR_VALUE)) 控制：是"，表示此变量支持用 Hint `SET_VAR()` 在运行时修改。 
+
 ## 变量参考
 
 ### `allow_auto_random_explicit_insert` <span class="version-mark">从 v4.0.3 版本开始引入</span>
 
 - 作用域：SESSION | GLOBAL
 - 是否持久化到集群：是
+- 受 Hint [SET_VAR](/optimizer-hints.md#SET_VAR(VAR_NAME=VAR_VALUE)) 控制：否
```
Comment (Oreoxmt) [inline system-variables.md:55]:
```suggestion
- 是否受 Hint [SET_VAR](/optimizer-hints.md#SET_VAR(VAR_NAME=VAR_VALUE)) 控制：否
```

File: optimizer-hints.md [hunk 6]
```diff
@@ -75,6 +75,17 @@ SELECT /*+ QB_NAME(QB1) */ * FROM (SELECT * FROM t) t1, (SELECT * FROM t) t2;
...
+### SET_VAR(VAR_NAME=VAR_VALUE)
+
+`SET_VAR(VAR_NAME=VAR_VALUE)` 能够将会话级系统变量，以 Hint 的形式在语句运行期间进行修改。语句执行完毕后，变量自动在所处的会话中恢复到原值。通过这个 Hint 可以修改一部分和优化器、执行器相关的变量行为，支持修改的系统变量请查看[系统变量](/system-variables.md)。强烈建议不要利用此 Hint 修改没有明确支持的变量，可能由此引发不可预知的行为。 
+
+```sql
+SELECT /*+ SET_VAR(MAX_EXECUTION_TIME=1234) */ @@MAX_EXECUTION_TIME;
+SELECT @@MAX_EXECUTION_TIME;
+```
+
+执行上述 SQL 会看到返回结果是 Hint 中设置的 1234，而不是变量 MAX_EXECUTION_TIME 的默认值。再次执行时，又恢复了默认值。
```
[the remark this answers is not in the document]
Comment (Oreoxmt) [reply optimizer-hints.md:87]: @winoros PTAL

File: optimizer-hints.md [hunk 7]
```diff
@@ -75,6 +75,38 @@ SELECT /*+ QB_NAME(QB1) */ * FROM (SELECT * FROM t) t1, (SELECT * FROM t) t2;
 >
 > 上述例子中，如果指定的 `QB_NAME` 为 `sel_2`，并且不给原本 `sel_2` 对应的第二个查询块指定新的 `QB_NAME`，则第二个查询块的默认名字 `sel_2` 会失效。
 
+### SET_VAR(VAR_NAME=VAR_VALUE)
+
+`SET_VAR(VAR_NAME=VAR_VALUE)` 允许在语句执行期间以 Hint 形式临时修改会话级系统变量的值。当语句执行完成后，系统变量将在当前会话中自动恢复为原始值。通过这个 Hint 可以修改一部分与优化器、执行器相关的系统变量行为。支持通过 `SET_VAR(VAR_NAME=VAR_VALUE)` Hint 修改的系统变量请查看[系统变量](/system-variables.md)。
+
+> **注意：**
```
Comment (ran-huang) [inline optimizer-hints.md:82]:
```suggestion
> **警告：**
```

File: system-variables.md [hunk 8]
```diff
@@ -42,12 +42,20 @@ SET GLOBAL tidb_distsql_scan_concurrency = 10;
 >
 > 单位相同的多个变量可能会争夺同一组资源。
 
+从 v7.4.0 开始，部分 `SESSION` 作用域的变量可以通过 [SET_VAR](/optimizer-hints.md#SET_VAR(VAR_NAME=VAR_VALUE)) Hint 在语句执行期间临时修改变量的值。当语句执行完成后，系统变量将在当前会话中自动恢复为原始值。通过这个 Hint 可以修改一部分与优化器、执行器相关的系统变量行为。文档中的变量有一个“是否受 Hint [SET_VAR](/optimizer-hints.md#SET_VAR(VAR_NAME=VAR_VALUE)) 控制”的说明，可以为“是”或者“否”。
```
Comment (ran-huang) [inline system-variables.md:45]:
```suggestion
从 v7.4.0 开始，部分 `SESSION` 作用域的变量可以通过 [`SET_VAR`](/optimizer-hints.md#SET_VAR(VAR_NAME=VAR_VALUE)) Hint 在语句执行期间临时修改变量的值。当语句执行完成后，系统变量将在当前会话中自动恢复为原始值。通过这个 Hint 可以修改一部分与优化器、执行器相关的系统变量行为。文档中的变量有一个“是否受 Hint [`SET_VAR`](/optimizer-hints.md#SET_VAR(VAR_NAME=VAR_VALUE)) 控制”的说明，可以为“是”或者“否”。
```

**Review by Oreoxmt — APPROVED** [review]

**Review by ran-huang — APPROVED** [review]

**Pull request merged**